### PR TITLE
BC-7473 - board type dialog

### PR DESCRIPTION
**Pull request by odalys-dataport**
# Short Description

Moving board create options from fab menu in to a dialog

## Links to Ticket and related Pull-Requests

BC-7473

## Screenshots of UI changes

Screenshot 2024-07-19 at 16 07 54

## Checklist before merging

- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/pages/rooms/RoomDetails.page.vue [hunk 1]
```diff
@@ -198,9 +208,29 @@ export default defineComponent({
 			isShareModalOpen: false,
 			isEndSyncDialogOpen: false,
 			tabIndex: 0,
+			boardTypePickerDialogIsOpen: false,
+			boardTypes: [
```
Comment (uidp) [inline src/pages/rooms/RoomDetails.page.vue:212]: `boardLayouts`?

File: src/pages/rooms/RoomDetails.page.vue [hunk 2]
```diff
@@ -198,9 +208,29 @@ export default defineComponent({
 			isShareModalOpen: false,
 			isEndSyncDialogOpen: false,
 			tabIndex: 0,
+			boardTypePickerDialogIsOpen: false,
```
Comment (uidp) [inline src/pages/rooms/RoomDetails.page.vue:211]: `boardLayoutPickerDialogIsOpen`

File: src/pages/rooms/RoomDetails.page.vue [hunk 3]
```diff
@@ -94,12 +94,19 @@
 			:course-id="roomData.roomId"
 			@success="refreshRoom"
 		/>
+		<SelectBoardLayoutDialog
+			v-if="boardLayoutsEnabled"
+			v-model="boardLayoutDialogIsOpen"
+			@select:multi-column="onMultiColumnLayoutSelected"
+			@select:single-column="onSingleColumnLayoutSelected"
+			@dialog-closed="boardLayoutDialogIsOpen = false"
```
Comment (uidp) [inline src/pages/rooms/RoomDetails.page.vue:102]: Is that extra event even necessary when we are using `v-model`? Shouldn't be the `boardLayoutDialogIsOpen` to `false` automatically via the two-way binding?

Comment (odalys-dataport) [reply src/pages/rooms/RoomDetails.page.vue:102]: Changed the close button click handler to the following

`@click="$emit('update:modelValue', false)"`

File: src/components/templates/DefaultWireframe.vue [hunk 4]
```diff
@@ -120,7 +120,7 @@ defineOptions({
 });
 const slots = useSlots();
 
-const isMobile = useVuetifyBreakpoints().smallerOrEqual("md");
+const isMobile = useVuetifyBreakpoints().smallerOrEqual("lg");
```
[the remark this answers is not in the document]
Comment (uidp) [reply src/components/templates/DefaultWireframe.vue]: Ok, I Changed the variable name.

File: src/modules/ui/speed-dial-menu/SpeedDialMenu.vue [hunk 5]
```diff
@@ -97,7 +97,7 @@ const actions = computed(() => {
 const isMenu = computed(() => actions.value.length > 0);
 const isMenuOpen = ref(false);
 
-const isMobile = useVuetifyBreakpoints().smallerOrEqual("md");
+const isMobile = useVuetifyBreakpoints().smallerOrEqual("lg");
```
[the remark this answers is not in the document]
Comment (uidp) [reply src/modules/ui/speed-dial-menu/SpeedDialMenu.vue]: see above.

**Review by MartinSchuhmacher — APPROVED** [review]

**Pull request merged**